### PR TITLE
Encapsulate pc_data

### PR DESCRIPTION
**Pull request by bluedreamer**

### CODE REVIEW

File: vme/src/modify.cpp [hunk 1]
```diff
@@ -1056,30 +1045,29 @@ void do_set(unit_data *ch, char *argument, const command_info *cmd)
             else
             {
                 slog(LOG_ALL, 0, "PASSWORD: %s changed %s's password.", UNIT_NAME(ch), UNIT_NAME(unt));
-                strncpy(PC_PWD(unt), crypt(strarg, PC_FILENAME(unt)), PC_MAX_PASSWORD);
-                PC_PWD(unt)[PC_MAX_PASSWORD - 1] = 0;
+                UPC(unt)->setPassword(crypt(strarg, PC_FILENAME(unt)));
```
Comment (Seifert69) [inline vme/src/modify.cpp:1048]: Double check that you chop it

File: vme/src/structs.h [hunk 2]
```diff
@@ -70,49 +70,402 @@ class pc_data : public char_data
...
+    }
+    int readPasswordFrom(CByteBuffer &buf)
+    {
+        auto retval = buf.ReadStringCopy(m_pwd, PC_MAX_PASSWORD);
+        m_pwd[PC_MAX_PASSWORD - 1] = '\0';
+        return retval;
+    }
+    void truncatePassword()
+    {
+        m_pwd[10] = '\0'; // This will allow me to later extend the password length
```
Comment (Seifert69) [inline vme/src/structs.h:394]: That 10 eventually needs addressing. Is it PC_MAX_PASSWORD.... ?

File: vme/src/modify.cpp (hunk 1 above)
Comment (bluedreamer) [reply vme/src/modify.cpp:1048]: Yes, it is done in the setter

File: vme/src/dilfld.cpp [hunk 3]
```diff
@@ -3676,25 +3687,28 @@ void dilfe_fld(dilprg *p)
...
                             v->atyp = DILA_NONE;
                             v->type = DILV_INT;
-                            v->val.num = PC_PROFESSION((unit_data *)v1->val.ptr);
+                            v->val.num = pc->getProfession();
                         }
                         else
                         {
                             v->type = DILV_UINT1R;
-                            v->ref = &PC_PROFESSION((unit_data *)v1->val.ptr);
+                            v->ref = pc->getPromptStringPtr();
```
Comment (Seifert69) [inline vme/src/dilfld.cpp:3703]: :-D   YaaaaY I found a bug :-)))

**Review by Seifert69 — CHANGES_REQUESTED** [review]
Really nice!! Thank you. You can ignore the one comment on password, it was fine. I did find one bug for you to fix just as I was beginning to think all my review efforts were futile :-D  I just almost didn't expand that file but then I thought, I'll do it anyway....

File: vme/src/modify.cpp (hunk 1 above)
Comment (Seifert69) [reply vme/src/modify.cpp:1048]: Agreed. It was a comment to myself but didn't go as intended :-)

File: vme/src/structs.h (hunk 2 above)
Comment (Seifert69) [reply vme/src/structs.h:394]: Yeah. I can't remember. There was definitely something tricky about the PWD. I believe that I increased it from 10 -> 13. So either I forgot the [10] or there was some obscure reason that I needed it to be 10.

File: vme/src/dilfld.cpp (hunk 3 above)
Comment (bluedreamer) [reply vme/src/dilfld.cpp:3703]: Great find!!

**Review by Seifert69 — APPROVED** [review]
Exciting :)

**Pull request merged**